src/wallet/rpcdump.cpp

```diff
@@ -622,3 +623,380 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
+            "      \"internal\": <true>                                    , (boolean, optional, default: false) Stating whether matching outputs should be be treated as not incoming payments\n"
+            "      \"watchonly\": <true>                                   , (boolean, optional, default: false) Stating whether matching outputs should be considered watched even when they're not spendable, only allowed if keys are empty\n"
+            "      \"label\": <label>                                      , (string, optional, default: '') Label to assign to the address (aka account name, for now), only allowed with internal=false\n"
+            "      \"timestamp\": 1454686740,                                (integer, optional, default now) Timestamp\n"
+            "    }\n"
+            "  ,...\n"
+            "  ]\n"
+            "2. json options                 (json, optional)\n"
+            "  {\n"
+            "     \"rescan\": <false>,         (boolean, optional, default: true) Stating if should rescan the blockchain after all imports\n"
+            "  }\n"
+            "\nExamples:\n" +
+            HelpExampleCli("importmulti", "'[{ \"scriptPubKey\": { \"address\": \"<my address>\" }, \"timestamp\":1455191478 }, "
+                                          "{ \"scriptPubKey\": { \"address\": \"<my 2nd address>\" }, \"label\": \"example 2\", \"timestamp\": 1455191480 }]'") +
+            HelpExampleCli("importmulti", "'[{ \"scriptPubKey\": { \"address\": \"<my address>\" }, \"timestamp\":1455191478 }]' '{ \"rescan\": false}'") +
+
+            "\nResponse is an array with the same size as the input that has the execution result :\n"
+            "  [{ \"success\": true } , { \"success\": false, \"error\": { \"code\": -1, \"message\": \"Internal Server Error\"} }, ... ]\n");
+    // clang-format on
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR)(UniValue::VOBJ));
+
+    const UniValue& request = params[0];
+
+    //Default options
+    bool fRescan = true;
+
+    if (params.size() > 1) {
+        const UniValue& options = params[1];
+        if (options.exists("rescan"))
+            fRescan = options["rescan"].get_bool();
+    }
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+    EnsureWalletIsUnlocked();
+
+    bool fRunScan = false;
+    int64_t nLowestTimestamp = 0;
+
+    if (fRescan && chainActive.Tip())
+        nLowestTimestamp = chainActive.Tip()->GetBlockTime();
+    else
+        fRescan = false;
+
+    UniValue response(UniValue::VARR);
+
+    BOOST_FOREACH (const UniValue& data, request.getValues()) {
+        const UniValue result = processImport(data);
+        response.push_back(result);
+    }
+
+    if (fRescan && fRunScan && request.size()) {
+        CBlockIndex* pindex = nLowestTimestamp > 0 ? chainActive.FindLatestBefore(nLowestTimestamp) : chainActive.Genesis();
```

Comment:
Are you sure this is correct?
I can't see a link between the input timestamp and `nLowestTimestamp`.
In case of a rescan, `nLowestTimestamp` is always `chainActive.Tip()->GetBlockTime()` which is incorrect IMO.

Also, we should add a time- or block-delta between `nLowestTimestamp` and the actual rescan height. Timestamps could be slightly wrong (timezones, failed implementations, accuracy). Maybe we should go down another 144 blocks.